File: Examples/VisualBasic/Data/AddOn/NamedRanges/MergeCellsInNamedRange.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells

Namespace Aspose.Cells.Examples.Data.AddOn.NamedRanges
    Public Class MergeCellsInNamedRange
        Public Shared Sub Main(ByVal args() As String)
            ' The path to the documents directory.
            Dim dataDir As String = Aspose.Cells.Examples.Utils.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Create directory if it is not already present.
            Dim IsExists As Boolean = System.IO.Directory.Exists(dataDir)
            If (Not IsExists) Then
                System.IO.Directory.CreateDirectory(dataDir)
            End If

            'Instantiate a new Workbook.
            Dim wb1 As New Workbook()

            'Get the first worksheet in the workbook.
            Dim worksheet1 As Worksheet = wb1.Worksheets(0)

            'Create a range.
            Dim mrange As Range = worksheet1.Cells.CreateRange("A18", "J18")

            'Name the range.
            mrange.Name = "Details"

            'Merge the cells of the range.
            mrange.Merge()

            'Get the range.
            Dim range1 As Range = wb1.Worksheets.GetRangeByName("Details")

            'Add a style object to the collection.
            Dim i As Integer = wb1.Styles.Add()

            'Define a style object.
            Dim style As Style = wb1.Styles(i)

            'Set the alignment.
            style.HorizontalAlignment = TextAlignmentType.Center

            'Create a StyleFlag object.
            Dim flag As New StyleFlag()
            'Make the relative style attribute ON.
            flag.HorizontalAlignment = True

            'Apply the style to the range.
            range1.ApplyStyle(style, flag)

            'Input data into range.
            range1(0, 0).PutValue("Aspose")

            'Save the excel file.
            wb1.Save(dataDir & "mergingrange.out.xls")
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Data/AddOn/NamedRanges/MergeCellsInNamedRange.cs
using System;
using System.IO;

using Aspose.Cells;

namespace Aspose.Cells.Examples.CSharp.Data
{
    public class MergeCellsInNamedRange
    {
        //Output directory
        static string outputDir = RunExamples.Get_OutputDirectory();

        public static void Run()
        {
            // Instantiate a new Workbook.
            Workbook wb1 = new Workbook();

            // Get the first worksheet in the workbook.
            Worksheet worksheet1 = wb1.Worksheets[0];

            // Create a range.
            Range mrange = worksheet1.Cells.CreateRange("D6", "I12");

            // Name the range.
            mrange.Name = "TestRange";

            // Merge the cells of the range.
            mrange.Merge();

            // Get the range.
            Range range1 = wb1.Worksheets.GetRangeByName("TestRange");      

            // Define a style object.
            Style style = wb1.CreateStyle();

            // Set the alignment.
            style.HorizontalAlignment = TextAlignmentType.Center;
            style.VerticalAlignment = TextAlignmentType.Center;
            style.Pattern = BackgroundType.Solid;
            style.ForegroundColor = System.Drawing.Color.Aqua;

            // Create a StyleFlag object.
            StyleFlag flag = new StyleFlag();
            // Make the relative style attribute ON.
            flag.HorizontalAlignment = true;
            flag.VerticalAlignment = true;
            flag.CellShading = true;

            // Apply the style to the range.
            range1.ApplyStyle(style, flag);

            // Input data into range.
            range1[0, 0].PutValue("Welcome to Aspose APIs.");

            // Save the excel file.
            wb1.Save(outputDir + "outputMergeCellsInNamedRange.xlsx");

            Console.WriteLine("MergeCellsInNamedRange executed successfully.");
        }
    }
}
